ErrorProne reports

```
/home/travis/build/java-json-tools/jackson-coreutils/src/main/java/com/github/fge/jackson/jsonpointer/TreePointer.java:171: warning: [EqualsGetClass] Overriding Object#equals in a non-final class by using getClass rather than instanceof breaks substitutability of subclasses.
    public final boolean equals(final Object obj)
                         ^
    (see https://errorprone.info/bugpattern/EqualsGetClass)
  Did you mean to remove this line?
/home/travis/build/java-json-tools/jackson-coreutils/src/main/java/com/github/fge/jackson/jsonpointer/TokenResolver.java:87: warning: [EqualsGetClass] Overriding Object#equals in a non-final class by using getClass rather than instanceof breaks substitutability of subclasses.
    public final boolean equals(final Object obj)
                         ^
    (see https://errorprone.info/bugpattern/EqualsGetClass)
  Did you mean to remove this line?
/home/travis/build/java-json-tools/jackson-coreutils/src/main/java/com/github/fge/jackson/jsonpointer/ReferenceToken.java:58: warning: [MutableConstantField] Constant field declarations should use the immutable type (such as ImmutableList) instead of the general collection interface type (such as List)
    private static final List<Character> ENCODED = ImmutableList.of('0', '1');
                             ^
    (see https://errorprone.info/bugpattern/MutableConstantField)
  Did you mean 'private static final ImmutableList<Character> ENCODED = ImmutableList.of('0', '1');'?
/home/travis/build/java-json-tools/jackson-coreutils/src/main/java/com/github/fge/jackson/jsonpointer/ReferenceToken.java:65: warning: [MutableConstantField] Constant field declarations should use the immutable type (such as ImmutableList) instead of the general collection interface type (such as List)
    private static final List<Character> DECODED = ImmutableList.of('~', '/');
                             ^
    (see https://errorprone.info/bugpattern/MutableConstantField)
  Did you mean 'private static final ImmutableList<Character> DECODED = ImmutableList.of('~', '/');'?
/home/travis/build/java-json-tools/jackson-coreutils/src/main/java/com/github/fge/jackson/JacksonUtils.java:96: warning: [MixedMutabilityReturnType] This method returns both mutable and immutable collections or maps from different paths. This may be confusing for users of the method.
    public static Map<String, JsonNode> asMap(final JsonNode node)
                                        ^
    (see https://errorprone.info/bugpattern/MixedMutabilityReturnType)
  Did you mean 'public static ImmutableMap<String, JsonNode> asMap(final JsonNode node)'?
```

--- a/src/main/java/com/github/fge/jackson/JacksonUtils.java
+++ b/src/main/java/com/github/fge/jackson/JacksonUtils.java
@@ -95,11 +95,11 @@ public final class JacksonUtils
      */
     public static Map<String, JsonNode> asMap(final JsonNode node)
     {
+        final Map<String, JsonNode> ret = Maps.newHashMap();
         if (!node.isObject())
-            return Collections.emptyMap();
+            return ret;
 
         final Iterator<Map.Entry<String, JsonNode>> iterator = node.fields();
-        final Map<String, JsonNode> ret = Maps.newHashMap();
 
         Map.Entry<String, JsonNode> entry;
 

--- a/src/main/java/com/github/fge/jackson/jsonpointer/ReferenceToken.java
+++ b/src/main/java/com/github/fge/jackson/jsonpointer/ReferenceToken.java
@@ -23,9 +23,9 @@ import com.github.fge.msgsimple.bundle.MessageBundle;
 import com.github.fge.msgsimple.load.MessageBundles;
 import com.google.common.collect.ImmutableList;
 
-import javax.annotation.concurrent.Immutable;
 import java.nio.CharBuffer;
 import java.util.List;
+import javax.annotation.concurrent.Immutable;
 
 
 /**
@@ -55,14 +55,14 @@ public final class ReferenceToken
     /**
      * List of encoding characters in a cooked token
      */
-    private static final List<Character> ENCODED = ImmutableList.of('0', '1');
+    private static final ImmutableList<Character> ENCODED = ImmutableList.of('0', '1');
 
     /**
      * List of sequences to encode in a raw token
      *
      * <p>This list and {@link #ENCODED} have matching indices on purpose.</p>
      */
-    private static final List<Character> DECODED = ImmutableList.of('~', '/');
+    private static final ImmutableList<Character> DECODED = ImmutableList.of('~', '/');
 
     /**
      * The cooked representation of that token

--- a/src/main/java/com/github/fge/jackson/jsonpointer/TokenResolver.java
+++ b/src/main/java/com/github/fge/jackson/jsonpointer/TokenResolver.java
@@ -90,7 +90,7 @@ public abstract class TokenResolver<T extends TreeNode>
             return false;
         if (this == obj)
             return true;
-        if (getClass() != obj.getClass())
+        if (!(obj instanceof TokenResolver))
             return false;
         final TokenResolver<?> other = (TokenResolver<?>) obj;
         return token.equals(other.token);

--- a/src/main/java/com/github/fge/jackson/jsonpointer/TreePointer.java
+++ b/src/main/java/com/github/fge/jackson/jsonpointer/TreePointer.java
@@ -29,9 +29,9 @@ import com.github.fge.msgsimple.load.MessageBundles;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Lists;
 
-import javax.annotation.concurrent.ThreadSafe;
 import java.util.Iterator;
 import java.util.List;
+import javax.annotation.concurrent.ThreadSafe;
 
 /**
  * A pointer into a {@link TreeNode}
@@ -174,7 +174,7 @@ public abstract class TreePointer<T extends TreeNode>
             return false;
         if (this == obj)
             return true;
-        if (getClass() != obj.getClass())
+        if (!(obj instanceof TreePointer))
             return false;
         final TreePointer<?> other = (TreePointer<?>) obj;
         return tokenResolvers.equals(other.tokenResolvers);

